diff --git a/dist/src/plugins/with-nx/with-nx.js b/dist/src/plugins/with-nx/with-nx.js
index ac6c3e960b1d463bbfd1c63339ffaed82b57f95a..97f1f4b9a64ba46bc98aa59c8fbb2ae4ed0f45ad 100644
--- a/dist/src/plugins/with-nx/with-nx.js
+++ b/dist/src/plugins/with-nx/with-nx.js
@@ -38,7 +38,7 @@ function ensureTypeScript() {
     }
     return ts;
 }
-function withNx(rawOptions, rollupConfig = {}, 
+function withNx(rawOptions, rollupConfig = {},
 // Passed by @nx/rollup:rollup executor to previous behavior of remapping tsconfig paths based on buildable dependencies remains intact.
 dependencies) {
     const finalConfig = { ...rollupConfig };
@@ -121,7 +121,7 @@ dependencies) {
     if (!rollupConfig.external && !global.NX_GRAPH_CREATION) {
         const npmDeps = (projectGraph.dependencies[projectNode.name] ?? [])
             .filter((d) => d.target.startsWith('npm:'))
-            .map((d) => d.target.slice(4));
+            .map((d) => d.target.slice(4).replace(/@[^@/]+$/, ''));
         let externalPackages = [
             ...Object.keys(packageJson.dependencies || {}),
             ...Object.keys(packageJson.peerDependencies || {}),
