diff --git a/src/plugins/with-nx/with-nx.js b/src/plugins/with-nx/with-nx.js
index 747921b173e0aa27030d1c8bd518248daf8885a9..957276f334845d3f1b614abc55f15a5bcc0de37b 100644
--- a/src/plugins/with-nx/with-nx.js
+++ b/src/plugins/with-nx/with-nx.js
@@ -39,7 +39,7 @@ function ensureTypeScript() {
     }
     return ts;
 }
-function withNx(rawOptions, rollupConfig = {}, 
+function withNx(rawOptions, rollupConfig = {},
 // Passed by @nx/rollup:rollup executor to previous behavior of remapping tsconfig paths based on buildable dependencies remains intact.
 dependencies) {
     const finalConfig = { ...rollupConfig };
@@ -122,7 +122,7 @@ dependencies) {
     if (!rollupConfig.external && !global.NX_GRAPH_CREATION) {
         const npmDeps = (projectGraph.dependencies[projectNode.name] ?? [])
             .filter((d) => d.target.startsWith('npm:'))
-            .map((d) => d.target.slice(4));
+            .map((d) => d.target.slice(4).replace(/@[^@/]+$/, ''));
         let externalPackages = [
             ...Object.keys(packageJson.dependencies || {}),
             ...Object.keys(packageJson.peerDependencies || {}),
