diff --git a/lib/util/version.js b/lib/util/version.js
index c9dc683b7d149c5a3a941bdcbd32f657facdfbf6..00b5ce2586c7c225cde0b458932482f3a8dea34e 100644
--- a/lib/util/version.js
+++ b/lib/util/version.js
@@ -28,7 +28,7 @@ function resetDetectedVersion() {
 
 function resolveBasedir(contextOrFilename) {
   if (contextOrFilename) {
-    const filename = typeof contextOrFilename === 'string' ? contextOrFilename : contextOrFilename.getFilename();
+    const filename = typeof contextOrFilename === 'string' ? contextOrFilename : (typeof contextOrFilename.getFilename === 'function' ? contextOrFilename.getFilename() : contextOrFilename.filename);
     const dirname = path.dirname(filename);
     try {
       if (fs.statSync(filename).isFile()) {
